Fails to configure with Perl 5.38.0(-RC0)

https://github.com/Perl/perl5/blob/blead/pod/perldelta.pod#changes-to-existing-diagnostics
^ The "when" keyword moves from "experimental" to "deprecated in Perl 5.38. 

This means it emits a warning, which triggers the `use warnings FATAL => 'all';` at the start of BuildCommon.pm.

```
checking for valgrind... no
checking for Python 3.>=6 with Passlib... not found
configure: Disabling the "regen-ka-table" target, missing Python requirements.
checking whether all ucontext.h functions are available... yes
when is deprecated at /var/tmp/portage/sys-libs/libxcrypt-4.4.34/work/libxcrypt-4.4.34/build-aux/scripts/BuildCommon.pm line 522.
Compilation failed in require at /var/tmp/portage/sys-libs/libxcrypt-4.4.34/work/libxcrypt-4.4.34/build-aux/scripts/expand-selected-hashes line 28.
BEGIN failed--compilation aborted at /var/tmp/portage/sys-libs/libxcrypt-4.4.34/work/libxcrypt-4.4.34/build-aux/scripts/expand-selected-hashes line 28.
configure: error: bad value 'all' for --enable-hashes
```

--- a/build-aux/scripts/BuildCommon.pm
+++ b/build-aux/scripts/BuildCommon.pm
@@ -11,7 +11,6 @@ use v5.14;    # implicit use strict, use feature ':5.14'
 use warnings FATAL => 'all';
 use utf8;
 use open qw(:utf8);
-no  if $] >= 5.018, warnings => 'experimental::smartmatch';
 no  if $] >= 5.022, warnings => 'experimental::re_strict';
 use if $] >= 5.022, re       => 'strict';
 
@@ -519,19 +518,19 @@ sub parse_symver_args {
     my $COMPAT_ABI;
     local $_;
     for (@args) {
-        when (/^SYMVER_MIN=(.+)$/) {
+        if (/^SYMVER_MIN=(.+)$/) {
             $usage_error->() if defined $SYMVER_MIN;
             $SYMVER_MIN = $1;
         }
-        when (/^SYMVER_FLOOR=(.+)$/) {
+        elsif (/^SYMVER_FLOOR=(.+)$/) {
             $usage_error->() if defined $SYMVER_FLOOR;
             $SYMVER_FLOOR = $1;
         }
-        when (/^COMPAT_ABI=(.+)$/) {
+        elsif (/^COMPAT_ABI=(.+)$/) {
             $usage_error->() if defined $COMPAT_ABI;
             $COMPAT_ABI = $1;
         }
-        default {
+        else {
             $usage_error->() if defined $map_in;
             $map_in = $_;
         }

